extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/taskadapter/PodTemplateTaskAdapter.java

```diff
@@ -243,16 +244,16 @@ private Map<String, String> getPodTemplateAnnotations(Task task) throws IOExcept
         .put(DruidK8sConstants.TASK_DATASOURCE, task.getDataSource())
         .build();
   }
-
+  
   private Map<String, String> getJobLabels(KubernetesTaskRunnerConfig config, Task task)
   {
     return ImmutableMap.<String, String>builder()
         .putAll(config.getLabels())
         .put(DruidK8sConstants.LABEL_KEY, "true")
-        .put(getDruidLabel(DruidK8sConstants.TASK_ID), task.getId())
-        .put(getDruidLabel(DruidK8sConstants.TASK_TYPE), task.getType())
-        .put(getDruidLabel(DruidK8sConstants.TASK_GROUP_ID), task.getGroupId())
-        .put(getDruidLabel(DruidK8sConstants.TASK_DATASOURCE), task.getDataSource())
+        .put(getDruidLabel(DruidK8sConstants.TASK_ID), KubernetesStringUtils.parseStringToK8sLabel(task.getId()))
+        .put(getDruidLabel(DruidK8sConstants.TASK_TYPE), KubernetesStringUtils.parseStringToK8sLabel(task.getType()))
+        .put(getDruidLabel(DruidK8sConstants.TASK_GROUP_ID), KubernetesStringUtils.parseStringToK8sLabel(task.getGroupId()))
+        .put(getDruidLabel(DruidK8sConstants.TASK_DATASOURCE), KubernetesStringUtils.parseStringToK8sLabel(task.getDataSource()))
```

Comment:
I think it is quite common for datasources to include `_` or `-`. The parseStringToK8sLabel function replaces these characters, even though they are valida characters in a k8s label because a task id can be very long in an attempt to fit more relevant information into the task_id label.

I think you should split `parseStringToK8sLabel` into 2 functions - one that just takes a string and makes it a k8s label value compliant string. And another function to deal with the task_ids